mongo/options/datakeyoptions.go

```diff
@@ -19,10 +19,37 @@ func DataKey() *DataKeyOptions {
 
 // SetMasterKey specifies a KMS-specific key used to encrypt the new data key.
 //
-// If being used with the AWS KMS provider, this option is required and must be a document with the following format:
-// {region: string, key: string}.
-//
 // If being used with a local KMS provider, this option is not applicable and should not be specified.
+//
+// For the AWS, Azure, and GCP KMS providers, this option is required and must be a document. For each, the value of the
+// "endpoint" or "keyVaultEndpoint" must be a host name with an optional port number (e.g. "foo.com" or "foo.com:443").
+//
+// When using AWS, the document must have the format:
+// {
+//   region: <string>,
+//   key: <string>,             // The Amazon Resource Name (ARN) to the AWS customer master key (CMK).
+//   endpoint: Optional<string> // An alternate host identifier to send KMS requests to.
+// }
+// If unset, the "endpoint" defaults to "kms.<region>.amazonaws.com".
+//
+// When using Azure, the document must have the format:
+// {
+//   keyVaultEndpoint: <string>,  // An alternate host identifier to send KMS requests to.
```

Comment:
This is not optional for Azure. I suggest removing  "alternate".

The docs I wrote for this were pretty slim: https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#masterkey-contents
> Required key vault endpoint. (e.g. "example.vault.azure.net")

https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#masterkey
> Host with optional port. Example: "example.vault.azure.net".

Though FWIW users familiar with Azure would probably know by the field name. Every resource in Azure is identified by a URL.

Reply:
Removed "alternate".